Add docs for caliban-quick uploads and derivation of fields from methods (#2045)
ghostdogpr/caliban: vuepress/docs/docs/schema.md

  </code-block>
</code-group>

### Deriving fields from case class methods (Scala 3 only)

> In certain cases, your type might contain fields of which their value depends on other fields. For example, you might have a `Person` type with a `fullName` field that is derived from the `firstName` and `lastName` fields. In this case, you can use the `@GQLField` annotation to indicate that the field should be derived from the method with the same name.

Reviewer: ```suggestion
In certain cases, your type might contain fields whose value depends on other fields. For example, you might have a `Person` type with a `fullName` field that is derived from the `firstName` and `lastName` fields. In this case, you can use the `@GQLField` annotation to indicate that the field should be derived from the method with the same name.
```

```scala
import caliban.schema.Schema
import caliban.schema.Annotations.GQLField

case class Person(
  firstName: String,
  lastName: String
) derives Schema.SemiAuto {
  @GQLField def fullName: String = s"$firstName $lastName"
}
```

This case class will generate the following GraphQL type:

```graphql
type Person {
  firstName: String!
  lastName: String!
  fullName: String!
}
```

The methods annotated with `@GQLField` can return any type for which a `Schema` is defined for, including effects such as `ZIO` and `ZQuery`.
In addition, you can use any other annotation that is supported for case class arguments, such as `@GQLName`, `@GQLDescription` and `@GQLDeprecated`.

::: warning Caveats

> Derivation of fields via the `@GQLField` annotation be convenient in certain cases, but has the following limitations:

Reviewer: ```suggestion
Derivation of fields via the `@GQLField` annotation can be convenient in certain cases, but has the following limitations:
```
